crates/interpreter/src/instructions/memory.rs

```diff
@@ -35,3 +42,35 @@ pub fn msize(interpreter: &mut Interpreter, _host: &mut dyn Host) {
     gas!(interpreter, gas::BASE);
     push!(interpreter, U256::from(interpreter.memory.effective_len()));
 }
+// From EIP-5656 MCOPY
+pub fn mcopy<SPEC: Spec>(interpreter: &mut Interpreter, host: &mut dyn Host) {
+    check!(interpreter, SPEC::enabled(CANCUN));
+    pop!(interpreter, dest, src, len);
+    if len == U256::ZERO {
+        return;
+    }
+
+    let len = as_usize_or_fail!(interpreter, len, InstructionResult::InvalidOperandOOG);
+    gas_or_fail!(interpreter, gas::verylowcopy_cost(len as u64));
+
+    // memory_resize!(interpreter, dest, len);
+    memory_copy_resize(interpreter, host);
+    let dest = as_usize_or_fail!(interpreter, dest, InstructionResult::InvalidOperandOOG);
+
+
+    let src = as_usize_or_fail!(interpreter, src, InstructionResult::InvalidOperandOOG);
+    // Read data with length len from src
+    let binding = interpreter.memory.copy_to_vec(src, len);
```

Comment:
It is not optimal to create temporary vec. There is a memory operation from `C` that does copying of overlapping data so I assume there is something similar in rust. Make `memory.copy` that does internal copying of data.

Reply:
I found and used `copy_within` 
https://doc.rust-lang.org/std/ptr/fn.copy.html

Reply:
This works perfectly